### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by fvanham-coveo**
Refresh of the readme which had gone stale. I removed the information on the fetch poly fills for node, it was no longer correct. Other text directly discussed Coveo capabilities and integrations, which are not really relevant to the readme. Other info is still there, just a little more compact.

Tagging some of the DP developers for context and a general intro to the library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,164 +1,181 @@
...
+        O.parentNode.insertBefore(u, O);
+    })(window, document, 'script', 'https://static.cloud.coveo.com/coveo.analytics.js/2/coveoua.js');
+</script>
 
-On top of `pageview` events, generic and commerce events can also be tracked. See [Tracking Commerce Events](https://docs.coveo.com/en/3188/coveo-solutions/tracking-commerce-events).
+coveoua('init', <COVEO_API_KEY>); // Replace <COVEO_API_KEY> with your api key</COVEO_API_KEY></COVEO_API_KEY>
+```
 
-**Note: This Coveo code snippet is similar to the Google analytics one (analytics.js).**
+Since calls to the coveo analytics service need to be authenticated, the library needs to be initialized with a Coveo api key which has push access to the [Analytics Data Domain](https://docs.coveo.com/en/1707/cloud-v2-administrators/privilege-reference#analytics-data-domain). You can create an API key from the [administration console](https://platform.cloud.coveo.com/admin/#/organization/api-access/) selecting the **Push** option box for the **Analytics Data** domain (see [Adding and Managing API Keys](https://docs.coveo.com/en/1718/cloud-v2-administrators/adding-and-managing-api-keys)).
```
Comment (samisayegh) [inline README.md:38]: Nice detail here 💯

File: README.md [hunk 2]
```diff
@@ -1,164 +1,181 @@
...
+-   `coveoua('init', <COVEO_API_KEY>, {endpoint: <ENDPOINT>, plugins: <PLUGINS>})`: Initializes the library with the given api key, endpoint and plugins. The following parameters are accepted
+    -   COVEO_API_KEY (mandatory): A valid api key.
+    -   ENDPOINT (optional): An object string specifying the desired analytics endpoint. The default value is https://analytics.cloud.coveo.com/rest/ua. In case your organization is HIPAA enabled, you should override with https://analyticshipaa.cloud.coveo.com/rest/ua.
+    -   PLUGINS (optional): An array of known plugin names. See [plugins](#plugins) for more information.
+-   `coveoua('set', <NAME>, <VALUE>)`: Attempts to inject an attribute with given name and value on every logged event. Some payloads may reject attributes they do not support.
+-   `coveoua('set', <OBJECT>)`: Attempts to inject all attributes and values of the given object on every logged event. Some payloads may reject attributes they do not support.
```
Comment (samisayegh) [inline README.md:55]: Would specifying an object override any previously specified name/value pairs? Perhaps worth mentioning if so,

File: README.md [hunk 3]
```diff
@@ -1,164 +1,181 @@
...
+
+-   Cookie storage, which supports top level domain storage. This means that the clientId for a.foo.com will be identical to the one on b.foo.com.
+-   Local storage, which allows to store much more information client side, but has the drawback of not being able to access data across multiple top level domains.
+-   Session storage, which has roughly the same limitation and capability as Local storage, except that it is cleared when the web browser tab is closed.
+
+By default, coveoua will use both local storage and cookie storage to persist it's clientId. If your environment does not support local persistence, it's possible to write your own storage abstraction.
+
+## Using coveo.analytics.js with React Native
+
+Since React Native does not run inside a browser, it cannot use cookies or the local/session storage that modern browsers provide. You must provide your own Storage implementation. Thankfully, there exist multiple packages to store data:
```
Comment (samisayegh) [inline README.md:171]: If using the library with react native becomes common, maybe an alternative could be to export the implementation for the developer to pass as a plugin?

**Review by samisayegh — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,164 +1,181 @@
...
-</script>
-```
+-   `event`: A generic event, which has been assembled through different plugin actions.
+-   `pageview`: An ecommerce specific pageview event which is automatically populated.
 
-To send commerce events, call `coveoua` with the event name. Here is how an [addition to the cart interaction](https://docs.coveo.com/en/3188/coveo-solutions/tracking-commerce-events#measuring-an-addition-to-the-cart) is measured:
+See the [Send an Event](https://docs.coveo.com/en/l3am0254/coveo-for-commerce/send-an-event) page for more information on the expected payloads for both of these.
 
-1. First use the `ec:addProduct` command to include the relevant product data in the event you’re about to send
+The full list of supported eCommerce plugin actions are listed in the [**Possible Actions** section of the Tracking Commerce Events page](https://docs.coveo.com/en/3188/coveo-for-commerce/commerce-data-health-implementation-guide#possible-actions). This page also provides information on how to log
```
Comment (GinoVin) [inline README.md:109]:
```suggestion
All supported actions are listed in the [**Product action type reference**](https://docs.coveo.com/en/l29e0540/coveo-for-commerce/commerce-events-reference#product-action-type-reference). This page also provides information on how to log
```

Comment (fvanham-coveo) [reply README.md:109]: This is a different list. eCommerce plugin actions are things like `ec:AddProduct` and `ec:addImpression`. I'm not sure if there a proper reference to all possible commands in the public doc. I can try to add them here.

Comment (GinoVin) [reply README.md]: You're right, we don't currently have an article that lists them all, they are however scattered throughout the documentation in the respective areas of usage.

**Review by lbourdages — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,164 +1,181 @@
...
+        u = o.createElement(v);
+        u.async = 1;
+        u.src = e;
+        O = o.getElementsByTagName(v)[0];
+        O.parentNode.insertBefore(u, O);
+    })(window, document, 'script', 'https://static.cloud.coveo.com/coveo.analytics.js/2/coveoua.js');
+</script>
 
-On top of `pageview` events, generic and commerce events can also be tracked. See [Tracking Commerce Events](https://docs.coveo.com/en/3188/coveo-solutions/tracking-commerce-events).
+coveoua('init', <COVEO_API_KEY>); // Replace <COVEO_API_KEY> with your api key</COVEO_API_KEY></COVEO_API_KEY>
```
[the remark this answers is not in the document]
Comment (fvanham-coveo) [reply README.md:35]: Thanks vscode! You're so helpful!

**Pull request merged**